wordpress-mobile/WordPress-Android
# Jetpack Restore: Display a warning for when server credentials are not configured

Fixes #14111,  #14666 (for Jetpack Restore)

Corresponding iOS PRs 
wordpress-mobile/WordPress-iOS/pull/15951, 
wordpress-mobile/WordPress-iOS/pull/16585

Corresponding FluxC PR: https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2017

To test:
1. Install the Jetpack app
2. Switch to a site with backup menu present + without server credentials configured.
3. Go to My Site > Backup > ellipsis icon > Restore.
4. Notice that
   1. The warning message is displayed. Also, there's a plus icon and only the `Enter your server credentials` is  highlighted.
   3. Clicking the message opens the Jetpack Settings for the selected site.
   3. The Restore site action button is disabled.
   4. Checkboxes are disabled.

Note: When server creds are configured, the message should not be displayed.

Server Creds not configured | Server Creds configured | Jetpack Settings 
-----------------------------|-------------------------|-----------------
![2](https://user-images.githubusercontent.com/1405144/120757887-c0475580-c52e-11eb-93c3-b7dc47ac85ab.png)| ![no_enter_server_creds_light](https://user-images.githubusercontent.com/1405144/120675527-c7c91900-c4b2-11eb-84e6-f09a409b4706.png) | ![enter_server_creds_light_settings](https://user-images.githubusercontent.com/1405144/120675578-d31c4480-c4b2-11eb-86b3-2c910e467909.png)

Merge Instructions:
1. Make sure FluxC PR is merged, create a tag and replace the temporary hash in the build.gradle 
2. Remove Not Ready for Merge, Needs Design Review labels
3. Merge the PR

## Regression Notes
1. Potential unintended areas of impact
WordPress: Restore option.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Tested the WP app manually. 

3. What automated tests I added (or what prevented me from doing so)
Tests added for this PR are applicable to both apps.

PR submission checklist:
